src/governance.ts

```diff
@@ -74,10 +74,13 @@ export class GovernanceCanister {
     }));
   };
 
-  public listProposals = async (
-    request: ListProposalsRequest,
-    certified = true
-  ): Promise<ListProposalsResponse> => {
+  public listProposals = async ({
+    request,
+    certified = true,
+  }: {
+    request: ListProposalsRequest;
+    certified?: boolean;
```

Comment:
When a value has a default, does the argument type represent what is required as an input - which may or may not have the value - or what is available within the function, which will always have a value?  I assume the former, yes?

Reply:
Yes, actually both. The argument type represent what is required or optional as an input and at the same time what's available since it's an interface.

By making `{certified?: boolean;}` it says "the parameter certified is optional"  and with `{certified = true}: {...}` it sets the default value

This is the same as `func(certified = true)` because you can call the func without params `func()` or with another value `func(false)`.

Does it answer your question?